src/domain/UploadToS3RecordRepository.ts

```diff
@@ -10,7 +11,7 @@ export type UploadToS3Record = {
   type: 'UploadToS3Record';
   input: {
     key: AmzDocumentKey;
-    checksumAlg: AmzSdkChecksumAlg;
+    checksumAlg: Option<AmzSdkChecksumAlg>;
```

Comment:
Try not to make structure embedding monads, just use types instead. 
If you want a structured types with optionals use io-ts instead.